api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -478,6 +495,136 @@ impl KubernetesInfrastructure {
+
+        for declared_volume in service.declared_volumes() {
+            let pvc_list_params = ListParams {
+                label_selector: Some(format!(
+                    "{}={},{}={},{}={}",
+                    APP_NAME_LABEL,
+                    app_name,
+                    SERVICE_NAME_LABEL,
+                    service.service_name(),
+                    STORAGE_TYPE_LABEL,
+                    declared_volume.split('/').last().unwrap_or("default")
+                )),
+                ..Default::default()
+            };
+            let fetched_pvc = existing_pvc.list(&pvc_list_params).await?.items;
+
+            if fetched_pvc.is_empty() {
+                let persistent_volume_claim = PersistentVolumeClaim {
+                    metadata: ObjectMeta {
+                        generate_name: Some(format!(
+                            "{}-{}-pvc-",
+                            app_name,
+                            service.service_name()
+                        )),
+                        labels: Some(BTreeMap::from([
+                            (APP_NAME_LABEL.to_owned(), app_name.to_owned()),
+                            (
+                                SERVICE_NAME_LABEL.to_owned(),
+                                service.service_name().to_owned(),
+                            ),
+                            (
+                                STORAGE_TYPE_LABEL.to_owned(),
+                                declared_volume
+                                    .split('/')
+                                    .last()
+                                    .unwrap_or("default")
+                                    .to_owned(),
+                            ),
+                        ])),
+                        ..Default::default()
+                    },
+                    spec: Some(PersistentVolumeClaimSpec {
+                        storage_class_name: storage_class.to_owned(),
+                        access_modes: Some(vec!["ReadWriteOnce".to_owned()]),
+                        resources: Some(ResourceRequirements {
+                            requests: Some(BTreeMap::from_iter(vec![(
+                                "storage".to_owned(),
+                                Quantity(
+                                    storage_size.to_owned().unwrap_or_else(|| "2Gi".to_owned()),
+                                ),
+                            )])),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                };
+
+                match Api::namespaced(client.clone(), app_name)
+                    .create(&PostParams::default(), &persistent_volume_claim)
+                    .await
+                {
+                    Ok(pvc) => {
+                        persistent_volume_map.insert(declared_volume, pvc);
+                    }
+                    Err(e) => {
```

Comment:
Is there a change to check if the PVC existed before? Because I'm not happy with treating it as an none error if there is something else wring.